### DISCUSSION TITLE
Generating a Graphql Schema

### DISCUSSION THREAD

**Discussion by azizbodal** [comunica/comunica#968] [Ideas]
I raised an issue in the [GraphQL-LD repo](https://github.com/rubensworks/GraphQL-LD.js/issues/39), but it was more of a question.

I was looking to create a graphql endpoint along with a graphiql UI. Is there a way to generate a graphql schema from the context or possibly the triple store (ex. the defined SHACL)?

**Comment by jeswr** [comment]
I've started work on SHACL schema generation in the past (though for typescript bindings rather than GraphQL schema) - see (https://github.com/SolidLabResearch/Challenges/issues/19#issuecomment-1072348932, https://www.researchgate.net/publication/348176302_on2ts_-Typescript_Generation_from_OWL_Ontologies_and_SHACL) and I believe others have done something similar with shex (e.g. https://github.com/o-development/shexj2typeandcontext). I've also investigated Construct query generation from SHACL (https://github.com/jeswr/shacl-query) - also a WIP.

Happy to talk through how you could use these as a starting point if you wanted to create a package of your own.

I'm not aware of any GraphQL specific schema generation work that has been done.

**Comment by azizbodal** [comment]
This is a great starting point! I'm sure I will have more questions as I
dig in further.

Thank you!
